web/ui/react-app/src/pages/Config.tsx

```diff
@@ -4,43 +4,47 @@ import { Alert, Button } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faSpinner } from '@fortawesome/free-solid-svg-icons';
 import CopyToClipboard from 'react-copy-to-clipboard';
-import { useFetch } from '../utils/useFetch';
 import PathPrefixProps from '../PathPrefixProps';
 
 import './Config.css';
+import { Fetcher, FetcherState } from '../api/Fetcher';
 
 const Config: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
-  const { response, error } = useFetch(`${pathPrefix}/api/v1/status/config`);
   const [copied, setCopied] = useState(false);
-
-  const config = response && response.data.yaml;
   return (
-    <>
-      <h2>
-        Configuration&nbsp;
-        <CopyToClipboard
-          text={config ? config! : ''}
-          onCopy={(text, result) => {
-            setCopied(result);
-            setTimeout(setCopied, 1500);
-          }}
-        >
-          <Button color="light" disabled={!config}>
-            {copied ? 'Copied' : 'Copy to clipboard'}
-          </Button>
-        </CopyToClipboard>
-      </h2>
+    <Fetcher url={`${pathPrefix}/api/v1/status/config`}>
```

Comment:
Extracting the children into their own component would make it even easier to test the display logic separately from the fetching, but maybe that's for a second step.

Reply:
Good point. I'm gonna think of how this can be done without introducing one more layer of components. Maybe a local functional component..